src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java

```diff
@@ -132,4 +151,50 @@ private QueryOuterClass.Query buildSelectQuery(DBFilterBase.IDFilter additionalI
         .limit(limit)
         .build();
   }
+
+  private QueryOuterClass.Query buildSortedSelectQuery(DBFilterBase.IDFilter additionalIdFilter) {
+    List<BuiltCondition> conditions = new ArrayList<>(filters.size());
+    for (DBFilterBase filter : filters) {
+      if (additionalIdFilter == null
+          || (additionalIdFilter != null && !(filter instanceof DBFilterBase.IDFilter)))
+        conditions.add(filter.get());
+    }
+    if (additionalIdFilter != null) {
+      conditions.add(additionalIdFilter.get());
+    }
```

Comment:
we should extract condition construction from the `buildSelectQuery` and use it here as well..